### ISSUE TITLE
[UX] Date fields: The form element for setting granularity is odd

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4650]
When configuring date fields, we currently allow for example to disable collecting the day and hour, while collecting minutes:

![image](https://user-images.githubusercontent.com/2423362/94336913-49e17380-ffef-11ea-96c2-ba0343526731.png)

I don't think that there's a legit use case for this, nor does it make any sense.

We should either:

1. Make it so that only the right-most _ticked_ granularity checkbox is enabled (and make sure that the checkboxes to its left are ticked and locked via `#states`)
2. Change this to a drop-down menu, similar to what is being proposed for the views UI in D9 in https://www.drupal.org/project/drupal/issues/2868014

   ![image](https://user-images.githubusercontent.com/2423362/94337036-22d77180-fff0-11ea-987b-447682625dee.png)
3. Ideally, we'd have something like #4649, so that we could change this to a range/slider form element (can be a follow-up)

PS: while at it, let's also add some help text to the field, because "granularity" is too technical. I believe that Date 7.x had "Select the date attributes to collect and store".

**Comment by indigoxela** [comment]
Agreed, the current granularity selection form item is odd and allows silly combinations.

**Comment by BWPanda** [comment]
What about something simple like: "Select the smallest date part" (wording TBA) with then a select list of options:

- Year
- Month
- Day
- Hour
- Minute
- Second

**Comment by indigoxela** [comment]
> ...with then a select list of options

Exactly my thinking.

**Comment by indigoxela** [comment]
Wow, it's impossible to solve [this](https://github.com/backdrop/backdrop/blob/5aba056ecbcaa20db3292fc70c0556fa1a178044/core/modules/date/date.admin.inc#L508)!

`$granularity` is supposed to be an array, changing that would need a bigger API change. See for instance [date_limit_format](https://github.com/backdrop/backdrop/blob/5aba056ecbcaa20db3292fc70c0556fa1a178044/core/includes/date.inc#L1163) in includes/date.inc.

Both, radios and select form items handle single values (in contrast to checkboxes items). Changing granularity setting back to array in the validation function is also impossible, because the form itself has a hook (date_field_settings_form). Besides the fact that all forms have alter hooks anyway.

The only thing that comes to my mind is a tricky and dirty solution with hidden fields, presenting the select list to users, but handling the checkboxes array in background. And that seems so ugly to me, that I don't even bother trying it.

**Comment by docwilmot** [comment]
Use JQuery to disable all except the checkbox to the left of the last unchecked box?

**Comment by jenlampton** [comment]
Yeah, let's go with @klonos's #1

> Make it so that only the right-most ticked granularity checkbox is enabled (and make sure that the checkboxes to its left are ticked and locked via #states)

**Comment by klonos** [comment]
> Yeah, let's go with @klonos's #1

That's one of the interesting this I'd like to work on ...I'll see if I can whip up a PR over this coming weekend, or if not, then soon as I can.
